examples/fs.go

```diff
@@ -103,8 +109,9 @@ func (d *Datastore) Query(q query.Query) (query.Results, error) {
 
 	walkFn := func(path string, info os.FileInfo, err error) error {
 		// remove ds path prefix
-		if strings.HasPrefix(path, d.path) {
-			path = path[len(d.path):]
+		relPath, err := filepath.Rel(d.path, path)
```

Comment:
(this may break in weird cases (.., etc.). but this is just an example.

Reply:
Yep, I think for the example datastore not worth thinking too hard about. If similar issues come up in FlatFS we'll deal with it there.